src/routes/[dictionaryId]/export.svelte

```diff
@@ -31,62 +35,83 @@
-      <span class="ml-2">xlsx</span>
-    </label> -->
-  </div>
-  <div class="flex items-center {hasImages ? '' : 'opacity-50 cursor-not-allowed'}">
-    <input disabled={!hasImages} id="images" type="checkbox" bind:checked={includeImages} />
-    <label for="images" class="mx-2 block leading-5 text-gray-900"> Images </label>
+    <input
+      disabled={!entriesWithImages.length}
+      id="images"
+      type="checkbox"
+      bind:checked={includeImages} />
+    <label for="images" class="mx-2 block leading-5 text-gray-900">
+      Images ({entriesWithImages.length})</label>
   </div>
   {#if !mounted}
-    <p class="text-xs italic text-orange-400 p-2">Checking if images are available</p>
-  {:else if !hasImages}
+    <p class="text-xs italic text-orange-400 p-2">Checking if images exist</p>
+  {:else if !entriesWithImages.length}
     <p class="text-sm text-red-700 p-3">There are no images</p>
   {/if}
-  <div class="flex items-center {hasAudio ? '' : 'opacity-50 cursor-not-allowed'}">
+
+  <div
+    class="flex items-center mt-2 {entriesWithAudio.length ? '' : 'opacity-50 cursor-not-allowed'}">
     <input id="audio" type="checkbox" bind:checked={includeAudio} />
-    <label for="audio" class="mx-2 block leading-5 text-gray-900"> Audio </label>
+    <label for="audio" class="mx-2 block leading-5 text-gray-900">
+      Audio ({entriesWithAudio.length})</label>
   </div>
   {#if !mounted}
-    <p class="text-xs italic text-orange-400 p-2">Checking if audio is available</p>
-  {:else if !hasAudio}
+    <p class="text-xs italic text-orange-400 p-2">Checking if audio files exist</p>
+  {:else if !entriesWithAudio.length}
     <p class="text-sm text-red-700 p-3">There are no audio files</p>
   {/if}
 </div>
 
-<Button
-  onclick={async () =>
-    await exportEntriesAsCSV(entries, $dictionary, { includeImages, includeAudio })}
-  form="primary">
-  Download CSV
-  {#if includeImages}
-    & Images
-  {/if}
-  {#if includeAudio}
-    & Audio
-  {/if}
-</Button>
+{#if includeImages || includeAudio}
+  <ShowHide let:show let:toggle>
+    {#if !show}
+      <Button onclick={toggle} form="primary">
+        Download CSV
+        {#if includeImages}
+          + Images
+        {/if}
+        {#if includeAudio}
+          + Audio
+        {/if}
+      </Button>
+    {:else}
+      <DownloadMedia
+        dictionary={$dictionary}
+        {formattedEntries}
+        entriesWithImages={includeImages ? entriesWithImages : []}
+        entriesWithAudio={includeAudio ? entriesWithAudio : []}
+        let:progress>
+        <Progress {progress} />
+        {#if progress < 1}
+          <Button onclick={toggle} color="red">Cancel</Button>
+        {:else}
+          <Button onclick={toggle}>Reset</Button>
+        {/if}
+      </DownloadMedia>
+    {/if}
+  </ShowHide>
+{:else}
+  <Button
+    disabled={!formattedEntries.length}
+    onclick={() => {
+      const finalizedEntries = formattedEntries.map((entry) => {
+        delete entry.impa;
```

Comment:
When I try to download only the CSV file but then I change my mind and try to download some media, this happens:
![image](https://user-images.githubusercontent.com/43384963/140169431-c3c1e522-f681-4d2c-9887-65c307a556b1.png)
I think it could be due to we are deleting the entry.impa and entry.aupa

Reply:
Good catch. I updated this to not change the original entry.